proxy-server/src/main/java/io/micrometer/prometheus/rsocket/PrometheusController.java

```diff
@@ -184,6 +190,26 @@ public Mono<String> prometheus() {
         .collect(Collectors.joining("\n"));
   }
 
+  private Duration determineTimeout(String timeoutHeader) {
+    if (timeoutHeader == null) {
+      return NO_TIMEOUT;
+    }
+
+    try {
+      Double timeoutMillis = Double.parseDouble(timeoutHeader) * 1E3;
+      Duration timeout = Duration
+          .ofMillis(timeoutMillis.longValue())
+          .minus(properties.getTimeoutOffset());
```

Comment:
[parseDouble](https://docs.oracle.com/javase/8/docs/api/java/lang/Double.html#parseDouble-java.lang.String-) returns a primitive `double`, the first line will autobox it, which might be unnecessary since `longValue` is just a cast to `long` under the hood.

You can do this: to convert the header to `Duration`:
```java
Duration.ofMillis((long) (Double.parseDouble(timeoutHeader) * 1_000));
```

Reply:
Good call, that's a bit simpler